ModTek/ProgressPanel.cs

```diff
@@ -52,12 +52,25 @@ public void SubmitWork(Func<IEnumerator<ProgressReport>> work)
             {
                 this.WorkList.AddLast(work);
             }
-
+            
             IEnumerator RunWorkList()
             {
                 foreach (var workFunc in WorkList) {
                     IEnumerator<ProgressReport> workEnumerator = workFunc.Invoke();
-                    while (workEnumerator.MoveNext())
+                   
+                    bool Next()
+                    {
+                        var sw = Stopwatch.StartNew();
+                        var didWork = false;
+                        var desiredFrameTime = 33.0;  // 30 fps - Steals a bit of cpu time from unity renderer
+
+                        // We want to work for the full duration of a frame, not one item per frame.
+                        while ((didWork = workEnumerator.MoveNext()) && sw.Elapsed.TotalMilliseconds < desiredFrameTime) { }
```

Comment:
< check is probably faster than object load to stack and function callout. do it first

Reply:
After our discussion, I remembered the reason for having it first.
We always want to advance the queue one item, no matter what. While it's silly and would "never happen", putting the time check first makes it a theoretical 0 or more, vs the desired 1 or more.